modules/templates/templates_plugin.rb

```diff
@@ -6,15 +6,8 @@ class Plugin < ::Proxy::Plugin
     plugin :templates, ::Proxy::VERSION
 
     after_activation do
-      unless Proxy::SETTINGS.foreman_url
-        logger.warn "missing :foreman_url: from configurations. 'templates' module is disabled"
-        ::Proxy::Plugins.disable_plugin(:templates)
-      end
-
-      unless Proxy::Templates::Plugin.settings.template_url
-        logger.warn "missing :templates_url: from configurations. 'templates' module is disabled"
-        ::Proxy::Plugins.disable_plugin(:templates)
-      end
+      loading_failed "missing :foreman_url: from configuration." unless Proxy::SETTINGS.foreman_url
+      loading_failed "missing :templates_url: from configuration." unless Proxy::Templates::Plugin.settings.template_url
```

Comment:
Looks like this should use `validate_presence` as it's in the plugin settings - it'll also check for empty values too then.  I don't think foreman_url can be checked like that, as it's a proxy/global setting.